src/kernel/components/multi_tasking.zig

```diff
@@ -71,9 +73,13 @@ pub const Process = struct {
             .memory_arena = std.heap.ArenaAllocator.init(ashet.memory.allocator),
             .name = undefined,
             .stay_resident = options.stay_resident,
+            .resources = undefined,
         };
         errdefer process.memory_arena.deinit();
 
+        process.resources = ashet.resources.HandlePool.init(process.memory_arena.allocator());
```

Comment:
Why not put this init call in the `process.*` assignment like the `memory_area` init?

Reply:
because it might alloc in the future and i want to keep a safe and refactorable  defer chain